client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java

```diff
@@ -92,6 +101,19 @@ public SortBasedPusher(
     this.taskAttemptId = taskAttemptId;
     this.numMappers = numMappers;
     this.numPartitions = numPartitions;
+
+    shuffledPartitions = new int[numPartitions];
+    reverseShuffledPartitions = new int[numPartitions];
+    for (int i = 0; i < numPartitions; i++) {
+      shuffledPartitions[i] = i;
+    }
+    if (conf.pushSortRandomizePartitionIdEnabled()) {
+      JavaUtils.randomizeInPlace(shuffledPartitions);
+      for (int i = 0; i < numPartitions; i++) {
+        reverseShuffledPartitions[shuffledPartitions[i]] = i;
```

Comment:
the "reverse" is confusing, is there terminology for such a calculation? cc @nafiyAix

Reply:
I can image "inverted", but I'm not sure

Reply:
we can call it "invertedShuffledPartitions"

https://www.pepcoding.com/resources/online-java-foundation/function-and-arrays/inverse_of_an_array/topic

Reply:
And we can assemble it in a method called "inverse", it's easy for testing and reusing